python/ray/dataframe/dataframe.py

```diff
@@ -506,7 +514,8 @@ def add_prefix(self, prefix):
         new_cols = self.columns.map(lambda x: str(prefix) + str(x))
         return DataFrame(block_partitions=self._block_partitions,
                          columns=new_cols,
```

Comment:
can we get rid of this line since we only need the metadata now?

Reply:
We can't in this case, as discussed today. If we don't pass the columns then the metadata of the new dataframe won't have the column changes reflected. Therefore, we need to either copy the metadata and modify the copy and push the copy, or pass the new columns such that the constructor modifies the metadata object copy on its end.